charcoal/utils.py

```diff
@@ -153,7 +153,12 @@ def gather_at_rank(mh, lca_db, lin_db, match_rank):
     # do the gather:
     counts = Counter()
     while 1:
```

Comment:
you could update this with `while query_sig.minhash:` and remove the first if statement.